src/win/process.c

```diff
@@ -1250,7 +1253,87 @@ static int uv__kill(HANDLE process_handle, int signum) {
+   */
+  if (signum == SIGQUIT) {
+    /* Get target process name. */
+    WCHAR basename[MAX_PATH];
+    GetModuleBaseNameW(process_handle, NULL, &basename[0], sizeof(basename));
+
+    /* Get PID of target process. */
+    DWORD pid = GetProcessId(process_handle);
+
+    /* Get LocalDumps directory path. */
+    HKEY registry_key;
+    DWORD ret = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                  L"SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps",
+                  0,
+                  KEY_QUERY_VALUE,
+                  &registry_key);
+    if (ret == ERROR_SUCCESS) {
+      WCHAR dump_folder[MAX_PATH];
+      DWORD dump_folder_len = sizeof(dump_folder);
+      DWORD key_type = 0;
+      ret = RegGetValueW(registry_key,
+                          NULL,
+                          L"DumpFolder",
+                          RRF_RT_ANY,
+                          &key_type,
+                          (PVOID) dump_folder,
+                          &dump_folder_len);
+      if (ret != ERROR_SUCCESS) {
+        /* Default value for `dump_folder` is `%LOCALAPPDATA%\CrashDumps`. */
+        WCHAR * localappdata;
+        SHGetKnownFolderPath(&FOLDERID_LocalAppData, 0, NULL, &localappdata);
+        _snwprintf_s(dump_folder, sizeof(dump_folder), _TRUNCATE, L"%ls\\CrashDumps", localappdata);
+        CoTaskMemFree(localappdata);
+      }
+      RegCloseKey(registry_key);
+
+      /* Create dump folder if it doesn't already exist. */
+      CreateDirectoryW(dump_folder, NULL);
+
+      /* Construct dump filename. */
+      WCHAR dump_name[MAX_PATH];
+      _snwprintf_s(dump_name, sizeof(dump_name), _TRUNCATE, L"%ls\\%ls.%d.dmp", dump_folder, basename, pid);
+
+      HANDLE hDumpFile = CreateFileW(dump_name, GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
+      if (hDumpFile != INVALID_HANDLE_VALUE) {
+        /* If something goes wrong while writing it out, delete the file to prevent clutter accumulation. */
+        FILE_DISPOSITION_INFO DeleteOnClose = { TRUE };
+        SetFileInformationByHandle(hDumpFile, FileDispositionInfo, &DeleteOnClose, sizeof(DeleteOnClose));
+
+        /* Tell wine to dump ELF modules as well. */
+        DWORD sym_options = SymGetOptions();
+        SymSetOptions(sym_options | 0x40000000);
+
+        if (MiniDumpWriteDump(process_handle, pid, hDumpFile,
+                              /*
+                              * We default to a rather large dump, so as to be more useful for debugging.
+                              * In the future, we may want to allow customization of what kinds of dumps
+                              * to create, as not all clients may wish to create such a large dump.
+                              */
+                              MiniDumpWithFullMemory | MiniDumpIgnoreInaccessibleMemory | MiniDumpWithAvxXStateContext,
+                              NULL, NULL, NULL) == TRUE) {
```

Comment:
style nit: each argument should usually get its own line. Might be worth moving this into a temporary variable also, but not strictly necessary (nor is `== TRUE` strictly necessary either, and a bit like adding `=== true` in julia to a boolean comparison)